athena-google-bigquery/src/main/java/com/amazonaws/athena/connectors/google/bigquery/BigQueryFederationExpressionParser.java

```diff
@@ -19,26 +19,112 @@
  */
 package com.amazonaws.athena.connectors.google.bigquery;
 
+import com.amazonaws.athena.connector.lambda.data.Block;
+import com.amazonaws.athena.connector.lambda.data.BlockUtils;
+import com.amazonaws.athena.connector.lambda.domain.predicate.Constraints;
+import com.amazonaws.athena.connector.lambda.domain.predicate.expression.ConstantExpression;
+import com.amazonaws.athena.connector.lambda.domain.predicate.expression.FederationExpression;
+import com.amazonaws.athena.connector.lambda.domain.predicate.expression.FunctionCallExpression;
+import com.amazonaws.athena.connector.lambda.domain.predicate.expression.VariableExpression;
 import com.amazonaws.athena.connector.lambda.domain.predicate.functions.FunctionName;
 import com.amazonaws.athena.connector.lambda.domain.predicate.functions.OperatorType;
 import com.amazonaws.athena.connector.lambda.domain.predicate.functions.StandardFunctions;
 import com.amazonaws.athena.connectors.jdbc.manager.FederationExpressionParser;
 import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
+import org.apache.arrow.vector.complex.reader.FieldReader;
+import org.apache.arrow.vector.types.Types;
 import org.apache.arrow.vector.types.pojo.ArrowType;
+import org.apache.arrow.vector.types.pojo.Field;
 import org.apache.commons.lang3.NotImplementedException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.amazonaws.athena.connector.lambda.domain.predicate.expression.ConstantExpression.DEFAULT_CONSTANT_EXPRESSION_BLOCK_NAME;
 
 /**
  * Based on com.amazonaws.athena.connectors.jdbc.manager.JdbcFederationExpressionParser class
  */
 public class BigQueryFederationExpressionParser extends FederationExpressionParser
```

Comment:
why does BQ need all of this extra logic?

Reply:
BQ does not used jdbc nor support prepare statement.  What team has done in the past is to create a plan BigQuery query. There is a bug on dealing with date in the existing logic

Reply:
we change the signature of FederationExpressionParser, hence we need to keep the old logic in BQ.